pkg/collector/corechecks/containers/docker/unbundled_events.go

```diff
@@ -0,0 +1,80 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build docker
+// +build docker
+
+package docker
+
+import (
+	"fmt"
+
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	"github.com/DataDog/datadog-agent/pkg/tagger/collectors"
+	"github.com/DataDog/datadog-agent/pkg/util/containers"
+	"github.com/DataDog/datadog-agent/pkg/util/docker"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+func newUnbundledTransformer(hostname string, types []string) eventTransformer {
+	collectedEventTypes := make(map[string]struct{}, len(types))
+	for _, t := range types {
+		collectedEventTypes[t] = struct{}{}
+	}
+
+	return &unbundledTransformer{
+		hostname:            hostname,
+		collectedEventTypes: collectedEventTypes,
+	}
+}
+
+type unbundledTransformer struct {
+	hostname            string
+	collectedEventTypes map[string]struct{}
+}
+
+func (t *unbundledTransformer) Transform(events []*docker.ContainerEvent) ([]metrics.Event, []error) {
+	var (
+		datadogEvs []metrics.Event
+		errors     []error
+	)
+
+	for _, ev := range events {
+		if _, ok := t.collectedEventTypes[ev.Action]; !ok {
+			continue
+		}
+
+		alertType := metrics.EventAlertTypeInfo
+		if ev.Action == "oom" || ev.Action == "kill" {
+			alertType = metrics.EventAlertTypeError
+		}
```

Comment:
should we extract this logic in a sub function? maybe also use `const string` to define "oom" and "kill"